### ISSUE TITLE
Enable stepwise processing flag for schedulers

### ISSUE THREAD

**Issue by peteriz** [PyTorchLightning/pytorch-lightning#806]
## 🚀 Feature
Asking if it makes sense adding a flag in the Trainer class for calling `scheduler.step()` after every update (per #640). 

### Motivation

This makes sense for training NLP models such as BERT/XLNet or any other that update the lr based on the current step (and training defined in terms of steps instead of epochs) instead of the current state that it is called after an epoch ends.
I'm aware that users can override `optimizer_step` of the model, however it's a quite common training pattern for training such NLP models.
I think this feature is worthwhile and I will contribute my changes. Let me know if not.

### Pitch

Add `scheduler.step()` call after every step (modify the optimizer step accordingly)

**Comment by williamFalcon** [comment]
instead of a flag, why not add the config to the scheduler? mostly because you might want to do different things to each  scheduler.

maybe allow a dictionary of options?

Here's an option:
```
def configure_optimizers(...):
    return [adam], [{'scheduler': Scheduler(), "step_frequency": "epoch|batch"}]
```

@neggert @jeffling
